Compound_Risk_database.R

```diff
@@ -870,56 +865,111 @@ socio_forward <- inform_covid_warning %>%
       TRUE ~ NA_real_
     ))
 
-#--------------------------Poverty projections----------------------------------------------------
+#--------------------------—MPO: Poverty projections----------------------------------------------------
 mpo <- read_dta("~/Google Drive/PhD/R code/Compound Risk/global.dta")
 
+# drive_download("Restricted_Data/global.dta", path = "tmp-global.dta", overwrite = T, verbose = F)
+# mpo <- read_dta("tmp-global.dta")
+# unlink("tmp-global.dta")
+
+# Add population
+pop <- wpp.by.year(wpp.indicator("tpop"), 2020)
+
+pop$charcode <- suppressWarnings(countrycode(pop$charcode,
+                                             origin = "iso2c",
+                                             destination = "iso3c"
+                                             )
+                                 )
+
+colnames(pop) <- c("Country", "Population")
+
 mpo_data <- mpo %>%
   rename(Country = Code) %>%
-  left_join(., pop, by= "Country") %>%
+  left_join(., pop, by= "Country")  %>%
   mutate_at(
     vars(contains("y20")),
-    funs( prop = . / Population)
+    ~ as.numeric(as.character(.))
   ) %>%
   mutate(
-    pov_prop_19_20 = ((y2020_prop / y2019_prop) -1) * 100,
-    pov_abs_19_20 = (y2020_prop - y2019_prop) * 100
-  ) %>%
+    pov_prop_22_21 = y2022 - y2021,
+    pov_prop_21_20 = y2021 - y2020,
+    pov_prop_20_19 = y2020 - y2019,
+    ) %>%
   filter(Label == "International poverty rate ($1.9 in 2011 PPP)") %>%
   rename_with(
     .fn = ~ paste0("S_", .),
     .cols = colnames(.)[!colnames(.) %in% c("Country")]
   ) 
 
-mpo_data <- normfuncpos(mpo_data, 50, 0, "S_pov_prop_19_20")
-mpo_data <- normfuncpos(mpo_data, 0.05, 0, "S_pov_abs_19_20")
+# Normalise based on percentiles
+mpo_data <- normfuncpos(mpo_data, 0.5, -1.5, "S_pov_prop_22_21")
+mpo_data <- normfuncpos(mpo_data, 1, -1, "S_pov_prop_21_20")
+mpo_data <- normfuncpos(mpo_data, 3, 0, "S_pov_prop_20_19")
+
+mpo_data <- mpo_data %>%
+  mutate(
+    S_pov_comb_norm = rowMaxs(as.matrix(dplyr::select(.,
+                                                      S_pov_prop_22_21_norm,
+                                                      S_pov_prop_21_20_norm, 
+                                                      S_pov_prop_20_19_norm))
+                              , na.rm = T)
+  )
+
+#-----------------------------—HOUSEHOLD HEATMAP FROM MACROFIN-------------------------------------
+# If Macro Fin Review is re-included above, we can reuse that. For clarity, moving data read here because it's not being used by macrosheet
+data <- read.csv("https://raw.githubusercontent.com/ljonestz/compoundriskdata/master/Indicator_dataset/macrofin.csv")
 
-#-----------------------------HOUSEHOLD HEATMAP FROM MACROFIN-------------------------------------
-household_risk <- macrosheet %>%
-  select(Country, M_Household.risks) %>%
+macrofin <- data %>%
```

Comment:
I've moved `macrofin` to the socioeconomic section because it is no longer used in macro, but is used for socioeconomic. Simple cut and paste to move.